tests/suite/test_requirements.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache 2.0 License.
+
+import infra.ccf
+import functools
+
+from loguru import logger as LOG
+
+
+class TestRequirementsNotMet(Exception):
+    pass
+
+
+def none(func):
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        return func(*args, **kwargs)
+
+    return wrapper
+
+
+# TODO: Parameterise this decorator once we add a test that requires a
+# different number of nodes
+def at_least_2_nodes(func):
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        if args[1].enforce_reqs is False:
+            return func(*args, **kwargs)
+
+        network = args[0]
+        running_nodes = len(network.get_joined_nodes())
+        if running_nodes < 2:
+            raise TestRequirementsNotMet(
+                f"Too few nodes. Only have {running_nodes}, requires 2"
+            )
+
+        return func(*args, **kwargs)
+
+    return wrapper
+
+
+def logging_app(func):
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        if args[1].enforce_reqs is False:
```

Comment:
If there's a way not to have this boilerplate in all the wrappers, please shout.

Reply:
A wrapper that makes this check that wraps all the wrappers??

Reply:
As discussed, there's no easy way to toggle the decorators, at least without writing more code than now.